src/server/scripts/EasternKingdoms/ShadowfangKeep/shadowfang_keep.cpp

```diff
@@ -205,6 +204,126 @@ class npc_arugal_voidwalker : public CreatureScript
+                std::copy(teleportSpells, teleportSpells + NUM_TELEPORT_SPELLS, _teleportSpells);
+            }
+
+            void KilledUnit(Unit* who) override
+            {
+                if (who->GetTypeId() == TYPEID_PLAYER)
+                    Talk(SAY_SLAY);
+            }
+
+            void SpellHitTarget(Unit* /*target*/, SpellInfo const* spell) override
+            {
+                if (spell->Id == SPELL_ARUGAL_CURSE)
+                    Talk(SAY_TRANSFORM);
+            }
+
+            void EnterCombat(Unit* /*who*/) override
+            {
+                _EnterCombat();
+                Talk(SAY_AGGRO);
+                events.ScheduleEvent(EVENT_CURSE, Seconds(7));
+                events.ScheduleEvent(EVENT_TELEPORT, Seconds(15));
+                events.ScheduleEvent(EVENT_VOID_BOLT, Seconds(1));
+                events.ScheduleEvent(EVENT_THUNDERSHOCK, Seconds(10));
+            }
+
+            void AttackStart(Unit* who) override
+            {
+                AttackStartCaster(who, 100.0f); // void bolt range is 100.f
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (!UpdateVictim())
+                    return;
+
+                events.Update(diff);
+
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                while (uint32 eventId = events.ExecuteEvent())
+                {
+                    switch (eventId)
+                    {
+                        case EVENT_CURSE:
+                            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 30.0f, true))
+                                DoCast(target, SPELL_ARUGAL_CURSE);
+                            events.Repeat(Seconds(15));
+                            break;
+                        case EVENT_TELEPORT:
+                        {
+                            // ensure we never cast the same teleport twice in a row
+                            uint8 spellIndex = urand(1, NUM_TELEPORT_SPELLS-1);
+                            std::swap(_teleportSpells[0], _teleportSpells[spellIndex]);
+                            DoCast(_teleportSpells[0]);
+                            events.Repeat(Seconds(20));
+                            break;
+                        }
+                        case EVENT_THUNDERSHOCK:
+                            DoCastAOE(SPELL_THUNDERSHOCK);
+                            events.Repeat(Seconds(30));
+                            break;
+                        case EVENT_VOID_BOLT:
+                            DoCastVictim(SPELL_VOIDBOLT);
+                            events.Repeat(Seconds(5));
+                            break;
+                    }
+
+                    if (me->HasUnitState(UNIT_STATE_CASTING))
+                        return;
+                }
+                DoMeleeAttackIfReady();
+            }
+
+            uint32 _teleportSpells[NUM_TELEPORT_SPELLS];
```

Comment:
don't call static and class field with the same name except a small _ difference, it's confusing and prone to errors.
doesn't it make more sense to avoid the whole static array idea and just initialize the class field to whatever you need to ?